plugin/src/middleware/request.ts

```diff
@@ -52,17 +53,35 @@ export class MiddlewareRequest {
 
   rewrite(destination: string | URL | NextURL, init?: ResponseInit): NextResponse {
     if (typeof destination === 'string' && destination.startsWith('/')) {
-      destination = new URL(destination, this.request.url)
+      destination = new URL(destination, this.url)
     }
     this.applyHeaders()
     return NextResponse.rewrite(destination, init)
   }
 
-  redirect(destination: string | URL | NextURL, init?: number | ResponseInit) {
-    if (typeof destination === 'string' && destination.startsWith('/')) {
-      destination = new URL(destination, this.request.url)
-    }
-    return NextResponse.redirect(destination, init)
+  get headers() {
+    return this.nextRequest.headers
+  }
+
+  get cookies() {
+    return this.nextRequest.cookies
+  }
+
+  get geo() {
+    return this.nextRequest.geo
+  }
+
+  get ip() {
+    return this.nextRequest.ip
+  }
+
+  get nextUrl() {
+    return this.nextRequest.url
+  }
+
+  get url() {
+    return this.nextRequest.url.toString()
```

Comment:
[dust] Would there be a situation where we'd want an `URL` instead of a `string` for `url`? An `URL` would be more flexible than a `string` depending on what the consumer of the property would be doing. Or maybe there's a reason it's only exposed as a string?

Reply:
This is a subclass of `Request`, so `url` will be a string.